amazon-eks-codebuild.yaml

```diff
@@ -0,0 +1,189 @@
+          - Effect: Allow
+            Action:
+              - "ec2:AttachVolume"
+              - "ec2:AuthorizeSecurityGroupIngress"
+              - "ec2:CopyImage"
+              - "ec2:CreateImage"
+              - "ec2:CreateKeypair"
+              - "ec2:CreateSecurityGroup"
+              - "ec2:CreateSnapshot"
+              - "ec2:CreateTags"
+              - "ec2:CreateVolume"
+              - "ec2:DeleteKeyPair"
+              - "ec2:DeleteSecurityGroup"
+              - "ec2:DeleteSnapshot"
+              - "ec2:DeleteVolume"
+              - "ec2:DeregisterImage"
+              - "ec2:DescribeImageAttribute"
+              - "ec2:DescribeImages"
+              - "ec2:DescribeInstances"
+              - "ec2:DescribeInstanceStatus"
+              - "ec2:DescribeRegions"
+              - "ec2:DescribeSecurityGroups"
+              - "ec2:DescribeSnapshots"
+              - "ec2:DescribeSpotPriceHistory"
+              - "ec2:DescribeSubnets"
+              - "ec2:DescribeTags"
+              - "ec2:DescribeVolumes"
+              - "ec2:DetachVolume"
+              - "ec2:GetPasswordData"
+              - "ec2:ModifyImageAttribute"
+              - "ec2:ModifyInstanceAttribute"
+              - "ec2:ModifySnapshotAttribute"
+              - "ec2:RegisterImage"
+              - "ec2:RunInstances"
+              - "ec2:StopInstances"
+              - "ec2:TerminateInstances"
+            Resource: "*"
+      Roles:
+        - !Ref CodeBuildRole
+
+  LogGroup:
+    Type: "AWS::Logs::LogGroup"
+    Properties:
+      LogGroupName: !Sub "/aws/codebuild/${Project}"
+      RetentionInDays: 3
+
+  Project:
+    Type: "AWS::CodeBuild::Project"
+    Properties:
+      Artifacts:
+        Type: NO_ARTIFACTS
+      Cache:
+        Modes:
+          - LOCAL_SOURCE_CACHE
+          - LOCAL_CUSTOM_CACHE
+        Type: LOCAL
+      Environment:
+        ComputeType: BUILD_GENERAL1_SMALL
+        EnvironmentVariables:
+          - Name: AWS_DEFAULT_REGION
+            Value: "us-west-2" # amazon-eks S3 bucket is in us-west-2
+          - Name: PACKER_VERSION
+            Value: "1.7.0"
+          - Name: PACKER_LOG
+            Value: !If
+              - IsPackerLog
+              - "1"
+              - "0"
+          - Name: PACKER_NO_COLOR
+            Value: "1"
+          - Name: CHECKPOINT_DISABLE
+            Value: "1"
+          - Name: K8S_VERSION
+            Value: !Ref KubernetesVersion
+        Image: aws/codebuild/standard:5.0 # AmazonLinux2 isn't supported by packer
+        Type: LINUX_CONTAINER
+      LogsConfig:
+        CloudWatchLogs:
+          Status: ENABLED
+      Name: amazon-eks-ami
+      Source:
+        BuildSpec: |-
+          version: 0.2
+
+          phases:
+            install:
+              commands:
+                - echo "Installing Packer..."
+                - curl -qL -o packer.zip https://releases.hashicorp.com/packer/${PACKER_VERSION}/packer_${PACKER_VERSION}_linux_amd64.zip && unzip packer.zip
```

Comment:
Can we replace this line with following and we don't need to pass packer version as environment variable. 
      - curl -fsSL https://apt.releases.hashicorp.com/gpg | apt-key add -
      - apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main"
      - apt-get update && apt-get install packer build-essential

Reply:
@awsvikrant I updated the PR, thanks for reviewing